docs/logging-usage.rst

```diff
@@ -0,0 +1,295 @@
+   >>> metric.exists()  # API call
+   False
+   >>> metric.create()  # API call
+   >>> metric.exists()  # API call
+   True
+
+List all metrics for a project:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> metrics, token = client.list_metrics()
+   >>> len(metrics)
+   0
+   >>> metric = metrics[0]
+   >>> metric.name
+   "robots"
+
+Refresh local information about a metric:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> metric = client.metric("robots")
+   >>> metric.get()  # API call
+   >>> metric.description
+   "Robots all up in your server"
+   >>> metric.filter
+   "log:apache-access AND textPayload:robot"
+
+Update a metric:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> metric = client.metric("robots")
+   >>> metric.exists()  # API call
+   True
+   >>> metric.get()  # API call
+   >>> metric.description = "Danger, Will Robinson!"
+   >>> metric.update()  # API call
+
+Delete a metric:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> metric = client.metric("robots")
+   >>> metric.exists()  # API call
+   True
+   >>> metric.delete()  # API call
+   >>> metric.exists()  # API call
+   False
+
+
+Export log entries using sinks
+------------------------------
+
+Sinks allow exporting entries which match a given filter to Cloud Storage
+buckets, BigQuery datasets, or Pubsub topics.
+
+Create a CloudStorage sink:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> sink = client.sink("robots-storage",
+   ...     filter='log:apache-access AND textPayload:robot')
+   >>> sink.storage_bucket = "my-bucket-name"
+   >>> sink.exists()  # API call
+   False
+   >>> sink.create()  # API call
+   >>> sink.exists()  # API call
+   True
+
+Create a BigQuery sink:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> sink = client.sink("robots-bq",
+   ...     filter='log:apache-access AND textPayload:robot')
+   >>> sink.bigquery_dataset = "projects/my-project/datasets/my-dataset"
+   >>> sink.exists()  # API call
+   False
+   >>> sink.create()  # API call
+   >>> sink.exists()  # API call
+   True
+
+Create a Pubsub sink:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> sink = client.sink("robots-pubsub",
+   ...     filter='log:apache-access AND textPayload:robot')
+   >>> sink.pubsub_topic = 'projects/my-project/topics/my-topic'
+   >>> sink.exists()  # API call
+   False
+   >>> sink.create()  # API call
+   >>> sink.exists()  # API call
+   True
+
+List all sinks for a project:
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> sinks, token = client.list_sinks()
+   >>> for sink in sinks:
+   ...     print(sink.name, sink.destination)
```

Comment:
Destination seems to be free-form (a bucket, a dataset, a topic, a man, a plan, a canal panama)